src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -112,21 +112,26 @@ public void playerJoin(PlayerJoinEvent event) {
         if (!p.hasPlayedBefore()) {
             this.plugin.log(Level.FINER, "Player joined for the FIRST time!");
             if (plugin.getMVConfig().getFirstSpawnOverride()) {
-                this.plugin.log(Level.FINE, "Moving NEW player to(firstspawnoverride): "
-                        + worldManager.getFirstSpawnWorld().getSpawnLocation());
+                this.plugin.log(Level.FINE, "Moving NEW player to firstspawnworld: " + worldManager.getFirstSpawnWorld().getSpawnLocation());
                 this.sendPlayerToDefaultWorld(p);
             }
             return;
-        } else {
+        }
+        else {
             this.plugin.log(Level.FINER, "Player joined AGAIN!");
+            if (this.plugin.getMVConfig().isJoinLocationOverride()) {
+                Logging.fine("Moving player to preset join location: " + this.plugin.getMVConfig().getJoinLocation());
+                this.sendPlayerToJoinLocation(p);
+            }
             if (this.plugin.getMVConfig().getEnforceAccess() // check this only if we're enforcing access!
                     && !this.plugin.getMVPerms().hasPermission(p, "multiverse.access." + p.getWorld().getName(), false)) {
                 p.sendMessage("[MV] - Sorry you can't be in this world anymore!");
                 this.sendPlayerToDefaultWorld(p);
             }
```

Comment:
It doesn't make sense to teleport a player to the join location, and then check if they're allowed to be there. We should probably check if they're allowed to be there, then teleport.